drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/FieldTypeProducer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.drools.workbench.screens.guided.dtable.client.widget.analysis;
+
+import java.util.Map;
+
+import org.drools.workbench.models.datamodel.oracle.ModelField;
+import org.drools.workbench.services.verifier.api.client.util.PortablePreconditions;
+import org.drools.workbench.services.verifier.plugin.client.api.FieldTypes;
+import org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracle;
+import org.kie.workbench.common.widgets.client.datamodel.AsyncPackageDataModelOracleImpl;
+
+public class FieldTypeProducer {
+
+    private AsyncPackageDataModelOracle oracle;
+
+    public FieldTypeProducer( final AsyncPackageDataModelOracle oracle ) {
+        this.oracle = PortablePreconditions.checkNotNull( "oracle",
+                                                          oracle );
+    }
+
+    public FieldTypes getFieldTypes() {
```

Comment:
No I think you are right. I use this to resolve field types, but really it is more. This should be something like getFactTypes or getModels.